pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -118,11 +124,168 @@ func (pl *VolumeBinding) EventsToRegister() []framework.ClusterEventWithHint {
+		"Pod", klog.KObj(pod),
+		"StorageClass", klog.KObj(modifiedStorageClass),
+	)
+
+	if oldStorageClass.AllowVolumeExpansion != modifiedStorageClass.AllowVolumeExpansion {
+		logger.V(4).Info("StorageClass was created or updated, and changed AllowVolumeExpansion", "AllowVolumeExpansion", modifiedStorageClass.AllowVolumeExpansion)
+		return framework.Queue, nil
+	}
+
+	if oldStorageClass.VolumeBindingMode != modifiedStorageClass.VolumeBindingMode {
+		logger.V(4).Info("StorageClass was created or updated, and changed Provisioner", "VolumeBindingMode", modifiedStorageClass.VolumeBindingMode)
+		return framework.Queue, nil
+	}
+
+	logger.V(4).Info("StorageClass was created or updated, but it doesn't make this pod schedulable")
+	return framework.QueueSkip, nil
+}
+
+func (pl *VolumeBinding) isSchedulableAfterPersistentVolumeClaimChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	oldPVC, modifiedPVC, err := util.As[*v1.PersistentVolumeClaim](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	logger = klog.LoggerWithValues(
+		logger,
+		"Pod", klog.KObj(pod),
+		"PersistentVolumeClaim", klog.KObj(modifiedPVC),
+	)
+
+	if pod.Namespace == modifiedPVC.Namespace {
+		for _, vol := range pod.Spec.Volumes {
+			var pvcName string
+			switch {
+			case vol.PersistentVolumeClaim != nil:
+				pvcName = vol.PersistentVolumeClaim.ClaimName
+			case vol.Ephemeral != nil:
+				pvcName = ephemeral.VolumeClaimName(pod, &vol)
+			}
+
+			if pvcName == modifiedPVC.Name {
+				if modifiedPVC.Status.Phase != oldPVC.Status.Phase {
+					logger.V(4).Info("PersistentVolumeClaim referenced by the Pod was created or updated, and changed Provisioner", "Phase", modifiedPVC.Status.Phase)
+					return framework.Queue, nil
+				}
+
+				if apiequality.Semantic.DeepEqual(modifiedPVC.Annotations, oldPVC.Annotations) {
+					logger.V(4).Info("PersistentVolumeClaim referenced by the Pod was created or updated, and changed Annotations")
+					return framework.Queue, nil
+				}
+
+				if apiequality.Semantic.DeepEqual(modifiedPVC.Spec, oldPVC.Spec) {
+					logger.V(4).Info("PersistentVolumeClaim referenced by the Pod was created or updated, and changed Spec")
+					return framework.Queue, nil
+				}
+			}
+		}
+	}
+
+	logger.V(4).Info("PersistentVolumeClaim was created or updated, but it doesn't make this pod schedulable")
+	return framework.QueueSkip, nil
+}
+
+func (pl *VolumeBinding) isSchedulableAfterPersistentVolumeChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	oldPV, modifiedPV, err := util.As[*v1.PersistentVolume](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	logger = klog.LoggerWithValues(
+		logger,
+		"Pod", klog.KObj(pod),
+		"PersistentVolume", klog.KObj(modifiedPV),
+	)
+
+	for _, vol := range pod.Spec.Volumes {
+		if vol.Name == modifiedPV.Name {
+			if apiequality.Semantic.DeepEqual(modifiedPV, oldPV) {
```

Comment:
If a PV was subject to processing by csi-translation-lib, it may undergo significant changes. Therefore, even if only a part of the PV changes, I believe it should be treated as a target for queueing. I would appreciate your thoughts on this.